integration/tests/lib/result_approval_state.go

```diff
@@ -42,34 +42,60 @@ func (r *ResultApprovalState) Add(sender flow.Identifier, approval *flow.ResultA
 // WaitForResultApproval waits until a result approval for execution result id from the verification node for
 // the chunk index within a timeout. It returns the captured result approval.
 func (r *ResultApprovalState) WaitForResultApproval(t *testing.T, verNodeID, resultID flow.Identifier, chunkIndex uint64) *flow.ResultApproval {
-	var resultApproval *flow.ResultApproval
+	approvals := r.WaitForTotalApprovalsFrom(t, flow.IdentifierList{verNodeID}, resultID, chunkIndex, 1)
+	if len(approvals) > 0 {
+		return approvals[0]
+	}
+	return nil
+}
+
+// WaitForTotalApprovalsFrom waits until "count" number of result approval for the given execution result id and
+// the chunk index is disseminated in the network from any subset of the given (verification) ids.
+// It returns the captured result approval.
+func (r *ResultApprovalState) WaitForTotalApprovalsFrom(
+	t *testing.T,
+	verificationIds flow.IdentifierList,
+	resultID flow.Identifier,
+	chunkIndex uint64,
+	count int) []*flow.ResultApproval {
+
+	receivedApprovalIds := flow.IdentifierList{}
+	receivedApprovals := make([]*flow.ResultApproval, 0)
+
 	require.Eventually(t, func() bool {
 		r.RLock()
 		defer r.RUnlock()
 
-		approvals, ok := r.resultApprovals[verNodeID]
-		if !ok {
-			return false
-		}
-
-		for _, approval := range approvals {
-			if !bytes.Equal(approval.Body.ExecutionResultID[:], resultID[:]) {
-				continue // execution result IDs do not match
+		for _, verificationId := range verificationIds {
+			approvals, ok := r.resultApprovals[verificationId]
+			if !ok {
+				return false
```

Comment:
```suggestion
				continue
```
We should still check for messages from other VNs in this case right?